pennylane/optimize/rotosolve.py

```diff
@@ -237,6 +242,12 @@ def cost_function(rot_param, layer_par, crot_param, rot_weights=None, crot_weigh
     With TensorFlow, a ``tf.Variable`` inside a ``tf.GradientTape`` may be used to
     mark variables as trainable.
 
+    .. warning::
+
+        ``RotosolveOptimizer`` will only update parameters that are *explicitly*
+        marked as trainable. Either via ``requires_grad`` if using Autograd or PyTorch,
+        or by using `tf.Variable` tensors inside a `GradientTape` if using TensorFlow.
```

Comment:
Although does this conflict with the warning below that the Rotosolve optimizer doesn't work with TF?

Reply:
I'm not entirely sure I understand the conflict, sorry. I'm seeing the above `..warning::` text and the text above the warning that states to use a GradientTape together with `tf.Variable` to mark parameters as trainable. Am I missing a warning?

Reply:
Oh I'm just thinking of this: https://github.com/PennyLaneAI/pennylane/issues/2083

Reply:
Very nice catch, I edited the warning and moved it further up in the docstring to hopefully catch all users coming across this incompatibility. :)